### PR TITLE
Fixed nodal swarm particle loss during advection snapback

### PR DESCRIPTION
**Pull request by jcgraciosa**
This fix mitigates the loss of nodal swarm particles during the snapback phase of SLCN advection.

### CODE REVIEW

File: src/underworld3/systems/ddt.py [hunk 1]
```diff
@@ -229,10 +229,29 @@ def update_pre_solve(
             # with self._nswarm_psi.access():
             #     print("1S:", self._nswarm_psi.swarmVariable.data, flush=True)
 
-            # restore coords (will call dm.migrate after context manager releases)
+            # additional steps for snapback routine
+            og_mig_type = uw.function.dm_swarm_get_migrate_type(self._nswarm_psi)   # get original migrate type
+            uw.function.dm_swarm_set_migrate_type(self._nswarm_psi, 0)              # set to Migrate BASIC
+            mig_type = uw.function.dm_swarm_get_migrate_type(self._nswarm_psi)
```
Comment (julesghub) [inline src/underworld3/systems/ddt.py]: Is this line still needed?

**Review by julesghub — CHANGES_REQUESTED** [review]
Looks great JC! Let me know

File: src/underworld3/systems/ddt.py (hunk 1 above)
Comment (julesghub) [reply src/underworld3/systems/ddt.py]: that is line 235

**Comment by jcgraciosa** [comment]
Thank you for pointing this out! 
It's a remnant of the tests I did and I've removed line 235.

**Review by julesghub — APPROVED** [review]

**Pull request merged**